src/actions/assetsActions.js

```diff
@@ -794,3 +796,24 @@ export const checkForMissedAssetsAction = () => {
     }
   };
 };
+
+export const fetchAllCrosschainBalances = () => {
+  return async () => {
+    // eslint-disable-next-line no-unused-vars
+    const { Mainnet, Matic, Bsc, Xdai } = NetworkNames;
+
+    const mainnetBalance = await etherspotService.instances[NetworkNames.Mainnet].getAccountBalances();
+    const maticBalance = await etherspotService.instances[NetworkNames.Matic].getAccountBalances();
+    const bscBalance = await etherspotService.instances[NetworkNames.Bsc].getAccountBalances();
+    const xdaiBalance = await etherspotService.instances[NetworkNames.Xdai].getAccountBalances();
```

Comment:
Had some issues trying to do this in a more loop-friendly way. Will consult with @poocart when appropriate.

Reply:
You could try with something like
```
// this could be declared outside of action function
const SUPPORETED_NETWORKS = [NetworkNames.Mainnet, NetworkNames.Matic, ...];

// ...
const networkBalance = {};
networks.forEach((network) => {
  networkBalance[network] = await etherspotService.instances[networ].getAccountBalances();
});
````

Reply:
^ looks good enough

Reply:
I'm not sure that `await`s inside a forEach loop work but i will try for sure

Reply:
Unfortunately doesn't work due to `forEach`'s async nature, but i alway's remember in Node land that the `for` loop worked but was a bit _dirty_... however i'll try to Promise.all the `networkBalance` first and remove the `await` as to just return a Promise.

Reply:
Sorted ✅

Reply:
You can do with `map` and `Promise.all`.

Reply:
>  You can do with map and Promise.all.

We have a winner!